src/components/WindowViewer.js

```diff
@@ -44,31 +54,63 @@ class WindowViewer extends Component {
    * @return [Boolean]
    */
   infoResponseIsInStore() {
-    const { infoResponses } = this.props;
-    const currentInfoResponse = infoResponses[this.imageInformationUri()];
-    return (currentInfoResponse !== undefined
-      && currentInfoResponse.isFetching === false
-      && currentInfoResponse.json !== undefined);
+    const responses = this.currentInfoResponses();
+    if (responses.length === this.currentCanvases().length) {
+      return true;
+    }
+    return false;
   }
 
   /**
-   * Constructs an image information URI to request from a canvas
+   * Figures out how many and what canvases to present to a user based off of
+   * the view, number of canvases, and canvasIndex.
    */
-  imageInformationUri() {
+  currentCanvases() {
     const { window } = this.props;
-    return `${this.canvases[window.canvasIndex].getImages()[0].getResource().getServices()[0].id}/info.json`;
+    switch (window.view) {
+      case 'book':
+        if ( // FIXME: There is probably better logic floating around out there to determine this.
+          this.canvases.length > 0 // when there are canvases present
+          && window.canvasIndex !== 0 // when the first canvas is not selected
+          && window.canvasIndex + 1 !== this.canvases.length // when the last canvas is not selected
+        ) {
+          // For an even canvas
+          if (window.canvasIndex % 2 === 0) {
+            return [this.canvases[window.canvasIndex - 1], this.canvases[window.canvasIndex]];
+          }
+          return [this.canvases[window.canvasIndex], this.canvases[window.canvasIndex + 1]];
+        }
+        return [this.canvases[window.canvasIndex]];
+      default:
+        return [this.canvases[window.canvasIndex]];
+    }
+  }
+
+  /**
+   * currentInfoResponses - Selects infoResponses that are relevent to existing
+   * canvases to be displayed.
+   */
+  currentInfoResponses() {
+    const { infoResponses } = this.props;
```

Comment:
Probably an opportunity here to extract method to selector, and potentially memoize. But holding off at the moment to reduce the diff.